[Feat] 필수 API 앤드포인트 구현

YoungjaeRo opened the pull request:
## 💡 작업 내용
- [x]  필수 API 앤드포인트 구현
- [x]  Swagger 설정
- [x]  응답통일

## 💡 자세한 설명
(가능한 한 자세히 작성해 주시면 도움이 됩니다.)
- [x]  Swagger 기본 설정
- [x]  응답통일 (ApiResponse 구현)
- [x] 장소(Commonplace) 관련 API 구현
- [x] 친구(FriendRelationship) 관련 API 구현
- [x] 이미지(Image) 관련 API 구현

## 📗 참고 자료 (선택)

## 📢 리뷰 요구 사항 (선택)

## ✅ 셀프 체크리스트
- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요?
- [x] Assignees, Reviewers, Labels 를 등록했나요?
- [ ] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [ ] 테스트는 잘 통과했나요?
- [x] 불필요한 코드는 제거했나요?

YoungjaeRo committed:
refactor: 파일 구조 재정의

YoungjaeRo committed:
feat: 응답통일(ApiResponse 클래스 도입)

YoungjaeRo committed:
refactor: 에러 처리 핸들러에 ApiResponse 적용

YoungjaeRo committed:
feat: Swagger 설정 및 SwaggerConfig 구현

YoungjaeRo committed:
feat: 장소(Commonplace) 관련 기본적인 API 구현 완료

YoungjaeRo committed:
feat: 이미지(Image) 관련 기본적인 API 구현 완료

supsup-hae commented on src/main/java/com/adit/backend/domain/commonPlace/controller/CommonPlaceController.java:
```diff
@@ -0,0 +1,66 @@
+
+@RestController
+@RequestMapping("/api/places")
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
```
👏🏻

supsup-hae commented on src/main/java/com/adit/backend/domain/commonPlace/controller/CommonPlaceController.java:
```diff
@@ -0,0 +1,66 @@
+@RestController
+@RequestMapping("/api/places")
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class CommonPlaceController {
```
CommonPlace의 로직은 차후 지도 개발시 수정될 부분이 있을것 같습니다!
- KakaoMap API와의 연동
- `CommonPlace` 는 최초로 등록된 장소에 대한 엔티티이기 때문에 별도의 CRUD 과정을 거치게 되는 데이터는 `UserPlace`가 될 것 같습니다

supsup-hae commented on src/main/java/com/adit/backend/domain/commonPlace/dto/CommonPlaceRequestDto.java:
```diff
@@ -0,0 +1,24 @@
+
+@Getter
+@Setter
+public class CommonPlaceRequestDto {
```
DTO의 역할을 생각했을때 데이터의 불변성을 지키고 데이터 값이 변형될 수 있는 가능성을 배제하고자 `Record` 클래스로 작성을 추천드립니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/commonPlace/dto/CommonPlaceResponseDto.java:
```diff
@@ -0,0 +1,33 @@
+    private String subCategory;
+    private String url;
+
+    public CommonPlaceResponseDto(CommonPlace place) {
```
`Record` 클래스로 전환하게되면 코드의 중복도 많이 개선이 될거라고 생각합니다! (생성자, Getter..)

supsup-hae commented on src/main/java/com/adit/backend/domain/commonPlace/service/CommonPlaceService.java:
```diff
@@ -0,0 +1,63 @@
+    // 새로운 장소 생성
+    public CommonPlace createPlace(CommonPlaceRequestDto requestDto) {
+        CommonPlace place = new CommonPlace();
+        place.setPlaceName(requestDto.getPlaceName());
```
- OCP원리가 위배되는 코드인것 같습니다! 하단부의 `updatePlace()`와 코드를 비교했을때 엔티티의 생성과 수정을 명확히 구분하기 힘들어질 수 도 있어요! `@Builder`를 이용하는게 좋을 것 같습니다
- `toEntity` 메서드를 이용하면 코드도 더 깔끔해질 것 같아요!

supsup-hae commented on src/main/java/com/adit/backend/domain/commonPlace/service/CommonPlaceService.java:
```diff
@@ -0,0 +1,63 @@
+import com.adit.backend.global.error.exception.BusinessException;
+import com.adit.backend.global.error.exception.GlobalErrorCode;
+
+import groovy.util.logging.Slf4j;
```
`@Slf4j`은 `lombok`에서 지원하는 기능이지 않나요? 라이브러리 명이 제가 사용하던것과 달라서 여쭤봅니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/friendRelationship/service/FriendRelationshipService.java:
```diff
@@ -0,0 +1,68 @@
+
+    public FriendRelationship sendFriendRequest(FriendRequestDto requestDto) {
+        User fromUser = userRepository.findById(requestDto.getFromUserId())
+                .orElseThrow(() -> new BusinessException("User not found", GlobalErrorCode.NOT_FOUND_ERROR));
```
멋진 코드!

supsup-hae commented on src/main/java/com/adit/backend/domain/image/controller/ImageController.java:
```diff
@@ -0,0 +1,53 @@
+@RestController
+@RequestMapping("/api/images")
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class ImageController {
```
`ImageController`도 이후 S3 연동시 변경되는 부분이 많을것 같습니다! 기본적인 틀을 짜주셔서 덕분에 나중에 작업할때 유용하게 사용할것 같습니다 :D

supsup-hae commented on src/main/java/com/adit/backend/global/error/exception/BusinessException.java:
```diff
@@ -1,4 +1,4 @@
-package com.adit.global.error.exception;
```
수정 감사합니다 :D

supsup-hae reviewed (commented):
수고하셨습니다! 요청사항을 반영해주셔서 검토하기 간편했어요 :D 기본적인 API는 작성해 주셔서 해당 틀에서 변경해서 작성하기 유용할 것 같습니다! 작성해 주신 내용을 기반으로 수정 사항을 몇 개 검토해야 할 것 같아서 수정 후 알려드리겠습니다.

supsup-hae committed:
:recycle: refactor: 도메인 중심의 패키지 구조로 재구성 (#8)
  - place: 장소 관련 도메인 (CommonPlace, PlaceStatistics)
  - user: 사용자 및 친구관계 도메인 (FriendRelationship)
  - notification : 사용자 알림 관련 도메인
  - image: 이미지 처리 관련 도메인
  - global: 공통 컴포넌트 (예외처리, API 응답)

supsup-hae committed:
:recycle: refactor: 라이브러리 의존성 변경 (#8)
- `Slf4j (groovy -> lombok)`

supsup-hae committed:
:recycle: refactor:  엔티티 기본 생성자 접근 제한 추가 (#8)
- 무분별한 엔티티 생성 방지를 위한 PROTECTED 레벨 설정

supsup-hae committed:
:recycle: refactor: DTO 클래스를 record로 변환 및 from 메소드 추가 (#8)

supsup-hae committed:
:recycle: refactor: CommonPlace API 응답 생성 패턴 개선 및 서비스 로직 수정 (#8)
- 생성자 직접 호출 방식을 정적 팩토리 메서드(`from`)로 변경
- 엔티티 `Builder` 패턴 적용
- 서비스 레이어 로직 개선

supsup-hae committed:
:recycle: refactor: Image API 응답 생성 패턴 개선 및 서비스 로직 수정 (#8)
- 생성자 직접 호출 방식을 정적 팩토리 메서드(`from`)로 변경
- 엔티티 `Builder` 패턴 적용
- 서비스 레이어 로직 개선

supsup-hae committed:
:recycle: refactor: FriendRelationship API 응답 생성 패턴 개선 및 서비스 로직 수정 (#8)
- 생성자 직접 호출 방식을 정적 팩토리 메서드(`from`)로 변경
- 엔티티 `Builder` 패턴 적용
- 서비스 레이어 로직 개선

supsup-hae committed:
:recycle: refactor: ErrorCode 인터페이스 구현 추가 (#8)
  - `getHttpStatus()` 메소드 구현
  - `getMessage()` 메소드 구현

supsup-hae committed:
:wrench: chore(config): 라이브러리 버전 호환성 및 환경설정 개선 (#8)
- 의존성 충돌 해결을 위한 `gradle.build` 수정
- `.env` 파일을 통한 설정값 관리 추가
- `gitignore` 에 `.env` 파일 명시

supsup-hae committed:
:truck: refactor: UserRepository 디렉토리 변경 (#8)

supsup-hae committed:
:recycle: refactor: `FriendRelationship`을 `Friendship`으로 이름 변경 (#8)

supsup-hae commented:
## 💡 작업 내용
- [x] 도메인 중심의 패키지 구조 재구성
- [x] 엔티티 및 DTO 클래스 개선
- [x] API 응답 패턴 개선
- [x] 환경 설정 관리 개선

## 💡 자세한 설명
**구조 개선**
- 도메인 중심의 패키지 구조로 재구성
- DTO 클래스를 record로 변환

**API 응답 패턴 개선**
- CommonPlace, Image, Friendship API 응답 생성 패턴 개선
- 정적 팩토리 메서드(from), Builder 패턴 적용

**설정 관리 개선**
- 라이브러리 버전 호환성 개선
- .env 파일을 통한 설정값 관리 추가

**네이밍 개선**
- FriendRelationship을 Friendship으로 이름 변경

supsup-hae reviewed (approved):
리뷰 확인 부탁드립니다!
